### DISCUSSION TITLE
Confused by : ${{ env.OPTIONS }}

### DISCUSSION THREAD

**Discussion by Drallas** [community/community#27126] [Actions]
<p>I’m trying to learn Github Actions using a repo with a <a href="https://github.com/marketplace/actions/toc-generator" rel="noopener nofollow ugc">TOC Generator</a> action.</p>
<p>It’s uses a set of Options:</p>
<pre><code class="lang-auto">          GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN }}
          TARGET_BRANCH_PREFIX: ${{ env.BRANCH_PREFIX }}
          FOLDING: ${{ env.TOC_FOLDING }}
          MAX_HEADER_LEVEL: ${{ env.TOC_MAX_HEADER_LEVEL }}
          TOC_TITLE: ${{ env.TOC_TITLE }}
          CREATE_PR: ${{ env.TOC_CREATE_PR }}
          TARGET_PATHS: ${{ env.TOC_TARGET_PATHS }}
</code></pre>
<p>Those Options are explained in the toc-generator documentation; but where and how do i define or supply the env.VALUES?! (myrepo/settings/environments?)</p>
<p>Also the action documentation though extensive doesn’t explain this to me!</p>
<p>Any help would be appreciated.</p>

**Comment by airtower-luna** [comment]
<p>The <code>env</code> context refers to environment variables. You can set them in the workflow file, or dynamically by writing to a special file. For details, see:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/actions/reference/environment-variables" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/actions/reference/environment-variables" target="_blank" rel="noopener">Environment variables - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/actions/reference/environment-variables" target="_blank" rel="noopener">//docs.github.com/en/actions/reference/environment-variables</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by Drallas** [comment]
<p>Tnx</p>
<p>setting directly in workflow.yml makes more sense to me "</p>
<blockquote>
<p>TOC_TITLE: ${{ env.TOC_TITLE }} -&gt; TOC_TITLE:  “My TOC Title”</p>
</blockquote>
<p>But ‘writing to a special file’ doesn’t make sense to me; even after reading the linked GitHub Doc.</p>
<p>I know I can set <strong>secrets</strong> and <strong>environment</strong> options on each repo, but I’m not sure where <strong>env.TOC_TITLE</strong> should point to!?</p>

**Comment by Drallas** [comment]
<p>for instance when I look at  :</p><aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/technote-space/toc-generator/blob/e0f463cec3ba22c686e5af432516bddc0c094dc4/.github/workflows/toc.yml" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/technote-space/toc-generator/blob/e0f463cec3ba22c686e5af432516bddc0c094dc4/.github/workflows/toc.yml" target="_blank" rel="noopener nofollow ugc">technote-space/toc-generator/blob/e0f463cec3ba22c686e5af432516bddc0c094dc4/.github/workflows/toc.yml</a></h4>
<pre><code class="lang-yml">on:
  pull_request:
    types: [opened, synchronize, reopened, closed]

name: TOC Generator

jobs:
  toc:
    if: github.event.pull_request.head.user.id == github.event.pull_request.base.user.id
    name: TOC Generator
    runs-on: ubuntu-latest
    timeout-minutes: 3
    steps:
      - uses: technote-space/load-config-action@v1
        with:
          CONFIG_FILENAME: workflow-settings.json, workflow-details.json
          IGNORE_WARNING: 'true'
      - uses: technote-space/toc-generator@gh-actions
        with:
          GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN }}
</code></pre>

  This file has been truncated. <a href="https://github.com/technote-space/toc-generator/blob/e0f463cec3ba22c686e5af432516bddc0c094dc4/.github/workflows/toc.yml" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I see many ${{ env.somestyff) references and I’m trying to understand where to put those values!</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Drallas" data-post="3" data-topic="157313">
<div class="title">
<div class="quote-controls"></div>
 Drallas:</div>
<blockquote>
<p>But ‘writing to a special file’ doesn’t make sense to me; even after reading the linked GitHub Doc.</p>
</blockquote>
</aside>
<p>You only need that if you want to generate a value <em>dynamically</em> while your workflow is running. In that case see <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable">Setting an environment variable</a>.</p>
<p>Otherwise you should be able to just use static values like this:</p>
<pre><code class="lang-yaml">      - uses: technote-space/toc-generator@gh-actions
        with:
          TOC_TITLE: 'your title here'
          # ... other options
</code></pre>
<p>In the example you linked it looks like the <code>technote-space/load-config-action</code> creates the environment variables.</p>

**Comment by Drallas** [comment]
<p>I’m a bit new to this but i’m starting to see it.<br>
The values are derived from <strong>workflow-settings.json / workflow-details.json</strong> <a href="https://github.com/technote-space/toc-generator/tree/master/.github" rel="noopener nofollow ugc">inside this folder</a>.</p>
<p>I found an interesting series about GitHub Actions including the env part on <a href="https://dev.to/mihinduranasinghe/introduction-to-github-actions-part-1-1d6p" rel="noopener nofollow ugc">Dev.to</a> to get me started.</p>
<p>Tnx again for your help!</p>
